dashboard/lib/services/script_seed.rb

```diff
@@ -247,7 +247,21 @@ def self.import_script(script_data)
       # Needed because we already have some Scripts with invalid names
       script_to_import.skip_name_format_validation = true
       Script.import! [script_to_import], on_duplicate_key_update: get_columns(Script)
-      Script.find_by!(name: script_to_import.name)
+
+      # activerecord-import doesn't trigger callbacks for imported models, and
+      # Scripts rely on the after_save hook to invoke `generate_plc_objects`,
+      # so we invoke it manually.
+      #
+      # see https://github.com/zdennis/activerecord-import#callbacks
+      #
+      # Note that we use activerecord-import extensively in the script seeding
+      # process, so we may end up needing to manually invoke these callbacks
+      # for more models than just Script, in which case we should probably
+      # reassess the pattern being used here.
```

Comment:
manually invoking these callbacks seems ok to me -- this is what the docs for activerecord-import say to do. what kind of other pattern would you envision here? i could see extracting a helper to import a set of objects and then run the callbacks, but I wouldn't think this is a reason to move off of activerecord-import.

Reply:
The "other pattern" I am _most_ interested in is using the built-in bulk insert and update operations that are coming in Rails 6: https://edgeapi.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-insert_all https://edgeapi.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-upsert_all